packages/shared/src/vault/__tests__/executeSwap.test.ts

```diff
@@ -274,13 +276,154 @@ describe(executeSwap, () => {
+    );
+  });
+
+  it('rejects if destination asset and chain are not valid', async () => {
+    await expect(
+      executeSwap(
+        {
+          amount: '1',
+          destAsset: Assets.DOT,
+          destChain: Chains.Bitcoin,
+          destAddress: TESTNET_BTC_ADDRESS,
+          srcAsset: Assets.ETH,
+          srcChain: Chains.Ethereum,
+        },
+        {
+          network: ChainflipNetworks.sisyphos,
+          signer: new VoidSigner('MY ADDRESS').connect({
+            getNetwork: () => Promise.resolve({ chainId: 11155111n }),
+          } as any),
+        },
+        { nonce: 1 },
+      ),
+    ).rejects.toThrow(
+      'invalid asset and chain combination: {"chain":"Bitcoin","asset":"DOT"}',
+    );
+  });
+
+  it('rejects an invalid destination address', async () => {
+    await expect(
+      executeSwap(
+        {
+          amount: '1',
+          destAsset: Assets.BTC,
+          destChain: Chains.Bitcoin,
+          destAddress: 'invalid-btc-address',
+          srcAsset: Assets.ETH,
+          srcChain: Chains.Ethereum,
+        },
+        {
+          network: ChainflipNetworks.sisyphos,
+          signer: new VoidSigner('MY ADDRESS').connect({
+            getNetwork: () => Promise.resolve({ chainId: 11155111n }),
+          } as any),
+        },
+        { nonce: 1 },
+      ),
+    ).rejects.toThrow(
+      'Address "invalid-btc-address" is not a valid Bitcoin" address',
+    );
+  });
+
+  it('rejects if the source chain is not an evm chain', async () => {
+    await expect(
+      executeSwap(
+        {
+          amount: '1',
+          destAsset: Assets.BTC,
+          destChain: Chains.Bitcoin,
+          destAddress: TESTNET_BTC_ADDRESS,
+          srcAsset: Assets.DOT,
+          srcChain: Chains.Polkadot,
+        },
+        {
+          network: ChainflipNetworks.sisyphos,
+          signer: new VoidSigner('MY ADDRESS').connect({
+            getNetwork: () => Promise.resolve({ chainId: 11155111n }),
+          } as any),
+        },
+        { nonce: 1 },
+      ),
+    ).rejects.toThrow('Chain Polkadot is not an evm chain');
+  });
+
+  it('rejects if the signer is connected to the wrong evm chain', async () => {
+    await expect(
+      executeSwap(
+        {
+          amount: '1',
+          destAsset: Assets.BTC,
+          destChain: Chains.Bitcoin,
+          destAddress: TESTNET_BTC_ADDRESS,
+          srcAsset: Assets.ETH,
+          srcChain: Chains.Ethereum,
+        },
+        {
+          network: ChainflipNetworks.sisyphos,
+          signer: new VoidSigner('MY ADDRESS').connect({
+            getNetwork: () => Promise.resolve({ chainId: 404n }),
+          } as any),
+        },
+        { nonce: 1 },
+      ),
+    ).rejects.toThrow(
+      'Signer is connected to unexpected evm chain (expected: 11155111, got: 404)',
+    );
+  });
+
+  it('rejects if the destination chain is not an evm chain for a call', async () => {
```

Comment:
we can only do ccm between evm chains?

Reply:
at the moment, yes. i think that we will support it for solana in the future, but i dont think that was implemented yet. i tried to verify and this is the event you see when you try to do a ccm swap to bitcoin:
<img width="545" alt="Screenshot 2024-03-21 at 15 31 54" src="https://github.com/chainflip-io/chainflip-sdk-monorepo/assets/13310795/9ce5f113-d484-41e9-9658-f7db2bdf99a6">

Reply:
this should be the place where the backend defines if a chain supports a ccm swap: https://github.com/chainflip-io/chainflip-backend/blob/b1f4da7641b0d66fb98c708df88fbbf5072c33d1/state-chain/primitives/src/chains.rs#L122

Reply:
Good to know! Thanks!